### ISSUE TITLE
Different data structures for load in mac-os and linux

### ISSUE THREAD

**Issue by kirnap** [JuliaImages/Images.jl#693]
When I load an image from the same source I got different data types in mac and linux, Namely,
Mac version:
```julia
julia> arr = load(img);
julia> typeof(arr)
Array{ColorTypes.RGB4{FixedPointNumbers.Normed{UInt8,8}},2}
```
Linux version:
```julia
julia> arr = load(img);
julia> typeof(arr)
Array{ColorTypes.RGB{FixedPointNumbers.Normed{UInt8,8}},2}
```
That difference creates a problem in permutedims function. Here is an example:
```julia
julia> typeof(c1)
ImageCore.ChannelView{FixedPointNumbers.Normed{UInt8,8},3,Array{ColorTypes.RGB4{FixedPointNumbers.Normed{UInt8,8}},2}}
julia> permutedims(c1, (3,2,1))
ERROR: DimensionMismatch("new array has 224 color channels, must have 3")
Stacktrace:
 [1] check_ncolorchan(::Array{ColorTypes.RGB4{FixedPointNumbers.Normed{UInt8,8}},2}, ::Tuple{Int64,Int64,Int64}) at /Users/omer/.julia/v0.6/ImageCore/src/colorchannels.jl:356
 [2] similar(::ImageCore.ChannelView{FixedPointNumbers.Normed{UInt8,8},3,Array{ColorTypes.RGB4{FixedPointNumbers.Normed{UInt8,8}},2}}, ::Type{FixedPointNumbers.Normed{UInt8,8}}, ::Tuple{Int64,Int64,Int64}) at /Users/omer/.julia/v0.6/ImageCore/src/colorchannels.jl:117
 [3] permutedims(::ImageCore.ChannelView{FixedPointNumbers.Normed{UInt8,8},3,Array{ColorTypes.RGB4{FixedPointNumbers.Normed{UInt8,8}},2}}, ::Tuple{Int64,Int64,Int64}) at ./permuteddimsarray.jl:116
```
here is my ad-hoc solution...
```julia
macfix = convert(Array{FixedPointNumbers.Normed{UInt8,8},3},  c1);
julia> permutedims(macfix, (3,2,1)); # works perfectly fine
```

**Comment by timholy** [comment]
The different structures are a consequence of what the underlying C libraries return.

The issue about `permutedims` is definitely "our" problem, and it's a really awful bug. Same as https://github.com/JuliaImages/ImageCore.jl/issues/48. I have a rewrite for 0.7 [here](https://github.com/JuliaImages/ImageCore.jl/pull/52) that should fix this, and if you're brave you could test it.

Fixing this on 0.6 is unfortunately a breaking change, and consequently I'm inclined to leave it like it is and do all my breaking with the upgrade to julia 0.7.

**Comment by rsrock** [comment]
Ugh, RGB4 vs RGB again. Will #52 affect QuartzImageIO when it lands?  I should make note of that.

**Comment by timholy** [comment]
I posted a much longer explanation of what's happening in https://github.com/JuliaImages/ImageCore.jl/pull/52. The bottom line is that if you were formerly using `reinterpret`, you're likely to have to change it to `reinterpretc`. If you were using `colorview`, then I'd guess there will be no code changes required.

If your tests explicitly tested the *type* of object returned from `load` operations, you'll likely have to change those. That was actually the most annoying part of https://github.com/JuliaImages/ImageCore.jl/pull/52, I had written a lot of such tests that I later came to regret. I switched most of them to just checking the eltype and dimensions.
